assets/js/components/ChartSourceController/index.jsx

```diff
@@ -1,81 +1,167 @@
+                />
+                <span className="ChartSourceController-text">{title}</span>
+            </label>
+        );
+    });
+
+    const {description} = toggle[source];
 
     return (
-      <label htmlFor={htmlId} {...{ key }} className="ChartSourceController-item">
-        <input
-          type="radio"
-          value={key}
-          id={htmlId}
-          name={id}
-          checked={key === source}
-          onChange={event => changeSource(event.target.value)}
-        />
-        <span className="ChartSourceController-text">{title}</span>
-      </label>
+        <div>
+            <div className="ChartSourceController-options">
+                {toggleItems}
+            </div>
+            <p className="ChartSourceController-description">{description}</p>
+        </div>
     );
-  });
-
-  const { description } = toggle[source];
-
-  return (
-    <div>
-      <div className="ChartSourceController-options">
-        {toggleItems}
-      </div>
-      <p className="ChartSourceController-description">{description}</p>
-    </div>
-  );
 };
 
 
-const Markup = ({ items, toggle, styling, changeSource, source, downloadText, barTypes }) => {
-  const { scale, color, rotated } = styling;
-  return (
-    <div className="ChartSourceController">
-      <BarChart {...{ barTypes, scale, color, rotated, items, downloadText, source }} />
-      {toggle && buildToggle({ source, toggle, changeSource })}
-    </div>
-  );
+const Markup = ({items, toggle, styling, changeSource, source, downloadText, barTypes}) => {
+    const {scale, color, rotated} = styling;
+    return (
+        <div className="ChartSourceController">
+            <BarChart {...{barTypes, scale, color, rotated, items, downloadText, source}} />
+            {toggle && buildToggle({source, toggle, changeSource})}
+        </div>
+    );
 };
 
 
 class ChartSourceController extends React.Component {
-  constructor(...props) {
-    super(...props);
-
-    const { initial, items } = this.props;
-
-    this.state = {
-      source: initial || Object.keys(items)[0],
-    };
-
-    this.events = {
-      changeSource: this.changeSource.bind(this),
-    };
-  }
-
-  changeSource(source) {
-    this.setState({ source });
-  }
-
-  render() {
-    const { items: rawItems, toggle, styling, downloadText, barTypes } = this.props;
-    const { source } = this.state;
-    const { changeSource } = this.events;
-    const items = rawItems[source];
-
-    return <Markup {...{ items, toggle, styling, source, changeSource, downloadText, barTypes }} />;
-  }
+    constructor(...props) {
+        super(...props);
+
+        const {initial, items, type} = this.props;
+        const source = initial || Object.keys(items)[0];
+
+        const barItems = this.getBarItems(this.props.items, source, type);
+        this.state = {
+            source: source,
+            barItems: barItems,
+            barTypes: this.props.barTypes
+        };
+        this.fetchActualExpenditureUrls(type);
```

Comment:
this is a good place to check the fetaure flag and only do the fetch if the feature is enabled